aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -421,8 +424,19 @@ internal class RealAWSCognitoAuthPlugin(
                 }
             },
             {
+                val signInOptions = if (options !is AWSCognitoAuthSignInOptions) {
+                    AWSCognitoAuthSignInOptions.builder().authFlowType(AuthFlowType.USER_SRP_AUTH).build()
```

Comment:
Creating a new object will override other properties in options.

Reply:
What properties are we talking about? Do we use them in the subsequent actions?

Reply:
metadata and userAttributes

Reply:
Moved the check to actions